### PR TITLE
Upgrade JS dependencies to latest versions

### PR DESCRIPTION
**Pull request by jefflloyd**
This will be easier to review commit-by-commit.

This PR upgrades all dependencies in `package.json` to their latest (or near-latest) versions. There were several breaking version changes that had to be applied as a result. I've added some inline comments to describe some of the larger changes.

I verified the site still worked by running `npm run dev` and validating that I could perform the basic actions (viewing various pages, hotkeys, searching for workflows, copying snippets, etc).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Use Node.js
         uses: actions/setup-node@v2
         with:
-          node-version: "17.x"
+          node-version: "20.x"
```
Comment (jefflloyd) [inline .github/workflows/ci.yml:17]: It's a little bizarre - CI specified 17.x, but our Vercel deployment was still using 14.x (it doesn't even support non-LTS versions). Now, they're both using 20.x.

File: components/WorkflowCard.tsx [hunk 2]
```diff
@@ -23,10 +23,8 @@ const WorkflowCard = ({ workflow, isSearchResult }: WorkflowCardProps) => {
 
   // Be default we should use the <a> tag for the cards so search crawlers can find the URLs.
   const LinkCard = ({ children }: { children: React.ReactNode }) => (
-    <Link href={destination}>
-      <a className={CARD_STYLE} onClick={trackClick}>
-        {children}
-      </a>
+    <Link href={destination} className={CARD_STYLE} onClick={trackClick}>
+      {children}
```
Comment (jefflloyd) [inline components/WorkflowCard.tsx:27]: `<Link>` components now render directly as `<a>` tags as of Next 13. See [this](https://nextjs.org/docs/pages/building-your-application/upgrading/version-13#link-component) for more details. I ran their suggested codemod to upgrade our usages.

File: components/search/CustomHits.tsx [hunk 3]
```diff
@@ -1,28 +1,21 @@
...
-  searchResults,
-  children,
-}: {
-  searchState: SearchState;
-  searchResults: SearchResults;
-  children: React.ReactNode;
-}) {
-  if (!searchState.query || searchState.query.length == 0) {
+function Hits({ children }: { children: React.ReactNode }) {
+  const { results } = useInstantSearch();
```
Comment (jefflloyd) [inline components/search/CustomHits.tsx:6]: `connectStateResults` was removed in `react-instantsearch` v7. They [recommend](https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/react/#connectstateresults) getting the same information about state/results from this `useInstantSearch()` hook instead.

File: components/search/CustomSearchBox.tsx [hunk 4]
```diff
@@ -44,4 +48,14 @@ function SearchBox({ refine }: { refine: Dispatch<SetStateAction<string>> }) {
   );
 }
 
+function connectSearchBox(Component: any) {
+  const SearchBox = (props: any) => {
+    const data = useSearchBox(props);
```
Comment (jefflloyd) [inline components/search/CustomSearchBox.tsx:53]: Similar story here - `useSearchBox` replaced `connectSearchBox`. This is lightly adapted from their [documentation](https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/react/#connectors).

File: lib/build-search.ts [hunk 5]
```diff
@@ -32,23 +32,20 @@ import { getSortedWorkflowsData } from "./workflows";
       process.env.ALGOLIA_SEARCH_ADMIN_KEY!
     );
 
-    // Initialize the index with the algolia index name we set up
-    const index = client.initIndex("workflow_specs");
-    
+    const indexName = "workflow_specs";
```
Comment (jefflloyd) [inline lib/build-search.ts:35]: In version 4 of `algoliasearch`, you called `initIndex` on a client, which would then use that index for all future calls. Now, you provide the index to those individual calls. See [here](https://www.algolia.com/doc/libraries/javascript/v5/upgrade/#removal-of-initindex) for more info.

File: pages/workflows/[id].tsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ import Layout from "../../components/layout";
 import WorkflowTags from "../../components/WorkflowTags";
 import { Argument, Workflow } from "warp-workflows";
 import { CopyIcon } from "../../components/icons/copy";
-import ReactTooltip from "react-tooltip";
+import { Tooltip } from "react-tooltip";
```
Comment (jefflloyd) [inline pages/workflows/[id].tsx:10]: Yet another major version upgrade with a slightly different API. See the [upgrade guide](https://react-tooltip.com/docs/upgrade-guide/changelog-v4-v5) for more details on these changes.

File: tailwind.config.js [hunk 7]
```diff
@@ -41,5 +41,4 @@ module.exports = {
       },
     },
   },
-  plugins: [require("@tailwindcss/line-clamp")],
```
Comment (jefflloyd) [inline tailwind.config.js:44]: This is included as of Tailwind v3.3, so we don't need to explicitly include it anymore.

**Review by acarl005 — COMMENTED** [review]
Nice! Thanks for doing this

**Review by acarl005 — APPROVED** [review]

**Pull request merged**